src/codegen/sdk/typescript/symbol.py

```diff
@@ -319,7 +325,17 @@ def _move_to_file(
 
         # =====[ Make a new symbol in the new file ]=====
         # This will update all edges etc.
-        file.add_symbol(self)
+        should_export = False
+
+        if self.is_exported or [
+            usage
+            for usage in self.usages
+            if usage.usage_symbol not in encountered_symbols
+            and not self.transaction_manager.get_transaction_containing_range(usage.usage_symbol.file.path, usage.usage_symbol.start_byte, usage.usage_symbol.end_byte, TransactionPriority.Remove)
```

Comment:
This should be moved into a noapidoc method or helper since we use it all over the place